### DISCUSSION TITLE
Suggestion: Global Configuration for Client Components in Next.js

### DISCUSSION THREAD

**Discussion by RitikShilpkar** [vercel/next.js#69607] [Help]
### Summary

Description:

In my Next.js project, I'm facing an issue where using React hooks in a component requires adding "use client" at the top of each file to avoid errors like:

` React Hook "useMyHook" cannot be called in an async function.
`
Adding "use client" to every component that needs hooks becomes repetitive. I'm looking for a way to configure Next.js globally so that all components can use hooks without needing the "use client" directive.

Is there a way to achieve this?

**Comment by RitikShilpkar** [comment]
@toolmantim @chrislloyd @benschwarz @Peeja

**Reply by Peeja** [reply]
Yes? Why did you ping me on this? Please don't spam random people from the internet.

**Comment by yashpriyam** [comment]
@Juneezee

**Comment by Juneezee** [comment]
> I'm looking for a way to configure Next.js globally so that all components can use hooks without needing the "use client" directive.

If you want all your components to be Client Components, you can create a root `layout.tsx` and make it a Client Component and with `"use client"`. 

```tsx 
"use client"

export default function Layout({ children }: { children: React.ReactNode }) {
  return (
    <html lang="en">
      <body>
        <main>{children}</main>
      </body>
    </html>
  )
}
```

You don't actually need to write `"use client"` in all your components file. Once a component has `"use client"`, all its child components are also Client Component, see https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs

> This means that by defining a "use client" in a file, all other modules imported into it, including child components, are considered part of the client bundle.
